### DISCUSSION TITLE
focus not reset when route changed

### DISCUSSION THREAD

**Discussion by BinarySenseiii** [vercel/next.js#65231] [Help]
### Summary

[RecordedVideo (3).webm](https://github.com/vercel/next.js/assets/68985954/39fdf00b-fd35-4cab-b950-4c0ec0e6c786)

i've noticed an issue with focus management during route changes. Upon the initial load, the focus correctly skips to the main navigation content, which is the desired behavior. However, when navigating between routes subsequently, the focus doesn't reset to the expected top-level component, causing it to miss out entirely  

here is the live [deployed Version]( https://personal-website-phi-five.vercel.app/)

can someone guide how to fix

**Comment by pavelbuchta** [comment]
It's caused by having the top navigation in the layout.tsx

Put the navigation from layout.tsx to page.tsx and it will work on route change.

**Reply by BinarySenseiii** [reply]
i think Screen reading software, on page load, starts reading the content of the page from the beginning. In Single Page Application, there is no page load after the initial one.

When the contents gets updated or routes changed with out a page load, screen reading software has no clue by default so its happening

**Reply by BinarySenseiii** [reply]
i've fixed it .. by creating one span now on every route change it will be focus by default

```tsx
const skipLinkRef = useRef<HTMLSpanElement>(null)
  
  useEffect(() => {
    if (skipLinkRef.current) {
      skipLinkRef.current.focus()
    }
  }, [pathname])

<span className="sr-only" ref={skipLinkRef} tabIndex={0}></span>
```
